FFCDemoPaymentService/Messaging/AmqpReceiver.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using Amqp;
+using Amqp.Framing;
+using FFCDemoPaymentService.Messaging.Actions;
+
+namespace FFCDemoPaymentService.Messaging
+{
+    public class AmqpReceiver<T>
+    {
+        private readonly IMessageAction<T> action;
+        private ReceiverLink receiver;
+
+        public AmqpReceiver(Session session, string queueName, IMessageAction<T> messageAction, int credit = 5)
+        {
+            this.action = messageAction;
+            CreateReceiver(session, queueName);
+            receiver.Start(credit, OnMessage);
+        }
+
+        public void Close()
+        {
+            receiver.Close();
+        }
+
+        private void CreateReceiver(Session session, string queueName)
+        {
+            Console.WriteLine($"Creating {queueName} receiver");
+            this.receiver = new ReceiverLink(
+                session,
+                $"{queueName}Receiver",
+                new Source() {Address = queueName},
+                null);
+        }
+
+
+        private void OnMessage(IReceiverLink link, Message message)
+        {
+            // the below extension method will not resolve for some reason, so having to use ToString.
+            // var messageBody = message.GetBody<string>();
```

Comment:
Don't think we need to keep the commented out code, the alternative approach is clear enough to understand usage and ToString() methods are fairly common so don't think this will throw anyone.